### PR TITLE
Replace tenant in queryparams with header

### PR DESCRIPTION
**Pull request by rohitsngh27**
# Resolves
https://jira.rax.io/browse/CERES-444

# What
We need to remove the query param of tenant from the apis and add it to request header as `X-Tenant`.

## How to test
Can be tested using junit and rest api calls.

### CODE REVIEW

File: src/main/java/com/rackspace/ceres/app/web/MetadataController.java [hunk 1]
```diff
@@ -56,28 +56,25 @@ public MetadataController(MetadataService metadataService, Environment environme
 
   @GetMapping("/metricNames")
   public Mono<List<String>> getMetricNames(
-      @RequestParam(name = "tenant") String tenantParam,
-      @RequestHeader(value = "#{appProperties.tenantHeader}", required = false) String tenantHeader) {
-    return metadataService.getMetricNames(tenantParam);
+      @RequestHeader(value = "#{appProperties.tenantHeader}") String tenantHeader) {
```
[the remark this answers is not in the document]
Comment (rohitsngh27) [reply src/main/java/com/rackspace/ceres/app/web/MetadataController.java:59]: By default required is equal to true only.

File: src/main/java/com/rackspace/ceres/app/web/SuggestApiController.java [hunk 2]
```diff
@@ -41,22 +41,23 @@ public SuggestApiController(SuggestApiService suggestApiService) {
   /**
    * This endpoint provides a means of implementing an auto-complete.
    *
-   * @param tenant tenant of the customer
+   * @param tenantHeader tenant of the customer
```
[the remark this answers is not in the document]
Comment (rohitsngh27) [reply src/main/java/com/rackspace/ceres/app/web/SuggestApiController.java:44]: `@param` is generic word for method arguments I guess.

File: src/test/java/com/rackspace/ceres/app/web/SuggestApiControllerTest.java [hunk 3]
```diff
@@ -18,20 +18,36 @@
...
+import com.rackspace.ceres.app.config.AppProperties;
+import com.rackspace.ceres.app.config.SuggestTypeEnumConverter;
 import com.rackspace.ceres.app.services.SuggestApiService;
 import java.util.Collections;
 import java.util.List;
 import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
```
[the remark this answers is not in the document]
Comment (rohitsngh27) [reply src/test/java/com/rackspace/ceres/app/web/SuggestApiControllerTest.java:30]: There were some unused imports there. Will remove those.

**Review by matsboman — APPROVED** [review]
LGTM

**Review by harageth — APPROVED** [review]

**Review by rajeshdasi — APPROVED** [review]

**Pull request merged**